src/views/ThreeWActivityForm/index.tsx

```diff
@@ -0,0 +1,839 @@
+            setFieldValue(undefined, 'districts');
+        },
+        [setFieldValue],
+    );
+
+    const activitiesBySector = useMemo(() => (
+        listToGroupList(
+            (value?.activities ?? []).filter((activity) => isDefined(activity.sector)),
+            (activity) => activity.sector ?? 0,
+            (activity, _, index) => ({
+                ...activity,
+                mainIndex: index,
+            }),
+        )
+    ), [value?.activities]);
+
+    const actionItemsBySector = useMemo(() => (
+        listToGroupList(
+            optionsResponse?.actions ?? [],
+            (action) => action.sector,
+            (action) => action,
+        )
+    ), [optionsResponse?.actions]);
+
+    const sectorOptionsMap = useMemo(() => (
+        listToMap(
+            optionsResponse?.sectors,
+            (sector) => sector.id,
+            (sector) => sector,
+        )
+    ), [optionsResponse?.sectors]);
+
+    const actionOptionsMap = useMemo(() => (
+        listToMap(
+            optionsResponse?.actions,
+            (action) => action.id,
+            (action) => action,
+        )
+    ), [optionsResponse?.actions]);
+
+    const handleStartDateChange = useCallback((newDate: string | undefined) => {
+        setValue((oldVal) => {
+            const status = calculateStatus(newDate, oldVal?.end_date);
+
+            return {
+                ...oldVal,
+                start_date: newDate,
+                status,
+            };
+        });
+    }, [setValue]);
+
+    const handleEndDateChange = useCallback((newDate: string | undefined) => {
+        setValue((oldVal) => {
+            const status = calculateStatus(oldVal?.start_date, newDate);
+
+            return {
+                ...oldVal,
+                end_date: newDate,
+                status,
+            };
+        });
+    }, [setValue]);
+
+    const handleSubmitClick = useCallback(() => {
+        beforeSubmitRef.current = false;
+
+        const submit = createSubmitHandler(
+            validate,
+            onErrorSet,
+            (valFromArgs) => {
+                const val = valFromArgs as FormFields;
+                const sectorValuesMap = listToMap(val?.sectors, (item) => item, () => true);
+                const finalValue = {
+                    ...val,
+                    activities: val?.activities?.map((activity) => ({
+                        ...activity,
+                        custom_supplies: listToMap(
+                            activity?.custom_supplies,
+                            (item) => item.supply_label,
+                            (item) => item.supply_value,
+                        ),
+                        supplies: listToMap(
+                            activity?.supplies,
+                            (item) => item.supply_action,
+                            (item) => item.supply_value,
+                        ),
+                    })).filter((activity) => sectorValuesMap?.[activity.sector]),
```

Comment:
We have to clear the activities values when sector is unticked. And we can remove this logic